gen/src/write.rs

```diff
@@ -1366,6 +1382,36 @@ fn write_rust_box_extern(out: &mut OutFile, ident: &Pair) {
     writeln!(out, "#endif // CXXBRIDGE1_RUST_BOX_{}", instance);
 }
 
+fn write_rust_sizeof_extern(out: &mut OutFile, element: &RustName) {
+    let instance = element.to_mangled(out.types);
+
+    writeln!(out, "#ifndef CXX_BRIDGE_RUST_SIZEOF_{}", instance);
+    writeln!(out, "#define CXX_BRIDGE_RUST_SIZEOF_{}", instance);
+    writeln!(out, "::std::size_t cxxbridge1$rust_sizeof${}() noexcept;", instance);
+    writeln!(out, "#endif // CXX_BRIDGE_RUST_SIZEOF_{}", instance);
+
+}
+
+fn write_rust_sizeof_impl(out: &mut OutFile, element: &RustName) {
+    let instance = element.to_mangled(out.types);
+    let typename = element.to_typename(out.types);
+
+    writeln!(out, "#ifndef CXX_BRIDGE_RUST_SIZEOF_IMPL_{}", instance);
+    writeln!(out, "#define CXX_BRIDGE_RUST_SIZEOF_IMPL_{}", instance);
+    out.begin_block(Block::Namespace("detail"));
+    writeln!(out, "template<>");
+    writeln!(out, "::std::size_t rust_sizeof<{}>() {{", typename);
```

Comment:
I am hesitant about over-reliance on the linker for resolving these template specializations. If the user winds up instantiating `rust::Slice` in C++ with some type that they shouldn't, in this implementation they would only find that out as a link error, not a compile error, due to the template declaration (without definition) `template <typename T> size_t rust_sizeof()` present in include/cxx.h. In general deferring errors from compile time to link time makes them harder to debug.

I found a different way to do this which doesn't use undefined template declarations and is able to surface all errors at compile time: see #595 + #597.